kolibri/content/models.py

```diff
@@ -64,7 +64,24 @@ class ContentNode(MPTTModel):
     tags = models.ManyToManyField(ContentTag, symmetrical=False, related_name='tagged_content', blank=True)
 
     title = models.CharField(max_length=200)
+
+    # the instance_id is used for mapping a node between kolibri and the
+    # content curation server. We can't use the raw id, since ids aren't
+    # guaranteed to be consistent across different content curation servers
+    # (once we have a distributed content curation server), and content ids may
+    # be the same across different nodes within the same channel (for student
+    # logging and analytics purposes.) Apart from generating the instance_id
+    # upon node creation, we also change the instance_id whenever we move or
+    # copy a node).
+    instance_id = UUIDField(default=uuid.uuid4, editable=False, unique=True)
```

Comment:
Yeah, it does add overhead, because of the indices as well -- I've observed this with some benchmarks around Morango stuff. But if we're using these IDs in the URLs etc, and hence querying using these IDs, we want to have indices on this field anyway.